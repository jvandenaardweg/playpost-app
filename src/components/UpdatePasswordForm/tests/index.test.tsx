import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { Props, UpdatePasswordForm } from '../index';

const onChangeTextHandler = jest.fn();
const onPressUpdatePasswordHandler = jest.fn();

const defaultProps: Props = {
  onChangeText: onChangeTextHandler,
  onPressUpdatePassword: onPressUpdatePasswordHandler,
  password: '',
  isLoading: false,
  isSuccess: false,
}

describe('UpdatePasswordForm', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = render(<UpdatePasswordForm {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }

      wrapper = render(<UpdatePasswordForm {...props} />);
    });

    it('should fire onChangeText when filling in the password TextInput', () => {
      const examplePassword = 'p@ssW0rd!';
      const passwordTextInput = wrapper.getByTestId('UpdatePasswordForm-TextInput-password');

      fireEvent.changeText(passwordTextInput, examplePassword);
      expect(onChangeTextHandler).toHaveBeenCalledWith('password', examplePassword);
    });

    it('should disable the buttons if isLoading is true', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }

      wrapper.update(<UpdatePasswordForm {...props} />);

      expect(wrapper.getByTestId('UpdatePasswordForm-Button-update').props.disabled).toBe(true);
    });

    it('should disable the buttons if isSuccess is true', () => {
      const props = {
        ...defaultProps,
        isSuccess: true
      }

      wrapper.update(<UpdatePasswordForm {...props} />);

      expect(wrapper.getByTestId('UpdatePasswordForm-Button-update').props.disabled).toBe(true);
    });
  });
});
