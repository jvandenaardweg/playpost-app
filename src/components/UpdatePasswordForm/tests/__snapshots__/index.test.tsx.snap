// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdatePasswordForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="UpdatePasswordForm"
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 14,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEEEEE",
            "bottom": 0,
            "height": 50,
            "justifyContent": "center",
            "position": "absolute",
            "right": 4,
            "width": 50,
            "zIndex": 1,
          }
        }
        testID="InputGroup-View-right-element-container"
      >
        <View
          accessible={true}
          clickable={true}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="InputGroupPassword-TouchableOpacity-toggle-password-visible"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 20,
                },
                Object {
                  "height": 20,
                  "width": 20,
                },
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="InputGroupPassword-TouchableOpacity-toggle-password-visible-icon"
          >
            ÔÖù
          </Text>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 15,
              "lineHeight": 23,
            },
            Object {
              "marginBottom": 4,
            },
          ]
        }
        testID="InputGroup-Text-label"
      >
        Your new password
      </Text>
      <TextInput
        autoCapitalize="none"
        autoFocus={false}
        blurOnSubmit={false}
        clearButtonMode="always"
        editable={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Your new super secret password"
        returnKeyType="done"
        secureTextEntry={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEEEEE",
            "borderRadius": 6,
            "fontFamily": "Inter-Regular",
            "fontSize": 15,
            "height": 50,
            "lineHeight": undefined,
            "paddingLeft": 14,
            "paddingRight": 14,
          }
        }
        testID="UpdatePasswordForm-TextInput-password"
        textContentType="password"
        value=""
      />
    </View>
    <View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="UpdatePasswordForm-Button-update"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Save new password
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
